src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/DigestServiceAuthorizationManager.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Net;
+using System.Security.Cryptography;
+using System.Security.Principal;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Description;
+using System.ServiceModel.Dispatcher;
+using System.Text;
+
+namespace WcfService.TestResources
+{
+    public abstract class DigestServiceAuthorizationManager : ServiceAuthorizationManager, IServiceBehavior
+    {
+        private string _realm;
+        private readonly TimeSpan _nonceValidityTime = TimeSpan.FromMinutes(1);
+
+        public DigestServiceAuthorizationManager(string realm)
+        {
+            if (string.IsNullOrEmpty(realm))
+            {
+                throw new ArgumentException("realm");
+            }
+
+            _realm = realm;
+        }
+
+        private bool Authorized(DigestAuthenticationState digestState, OperationContext operationContext, ref Message message)
+        {
+            object identitiesListObject;
+            if (!operationContext.ServiceSecurityContext.AuthorizationContext.Properties.TryGetValue("Identities",
+                out identitiesListObject))
+            {
+                identitiesListObject = new List<IIdentity>(1);
+                operationContext.ServiceSecurityContext.AuthorizationContext.Properties.Add("Identities", identitiesListObject);
+            }
+            var identities = identitiesListObject as IList<IIdentity>;
+            identities.Add(new GenericIdentity(digestState.Username, "GenericPrincipal"));
```

Comment:
If the TryGetValue fails, then it will run the code inside the if block which ensures the identitiesListObject is not null.